src/symmetry.jl

```diff
@@ -314,3 +314,14 @@ function unfold_bz(scfres)
     new_scfres = (; basis=basis_unfolded, ψ, ham, eigenvalues, occupation)
     merge(scfres, new_scfres)
 end
+
+# symmetrize a rank-2 tensor in reduced coordinates
+function symmetrize_tensor(tensor, symmetries, lattice)
```

Comment:
In this file we generally follow the convention (system arguments, data arguments), i.e. it should be `lattice, symmetries, tensor`. 

Also can you add `::Matrix` or even better ::Mat3 to indicate this only works for matrices instead of general tensors.
In fact `::Mat3` would even allow to rename this function to `symmetrize` (same as the symmetrize we use for the densities) and add another method `symmetrize(model, tensor::Mat3)` and/or `symmetrize(model, tensor::Mat3)`.

Reply:
Not too sure about this, for two reasons. The first is that this routine is specific to stresses-like tensors, ie that transform in a specific way, being the derivative of the energy wrt the lattice, which transforms in another way (I would  imagine the term is that lattices are covariant, while stresses are contravariant? I'm hesitant here, this is definitely not my zone of confort). The second is that is that it's also specific to the fact that our stresses are in reduced coordinates. So I would call it `symmetrize_stresses`.

Unless it's canonical that "tensors" transform in this particular way? In which call calling it symmetrize_tensor is OK by me, since things are in reduced coordinates by default (we use `_cart` for non-reduced)

Reply:
To clarify with "about this" you mean the renaming to `symmetrize`, but not the rest.

Yeah ok, I might have been a bit quick here. From the way stresses are formed I agree with your argument that they could be special (likely the only contravariant thing we deal with in the foreseeable time). I also never looked into this in detail, however.

Reply:
> To clarify with "about this" you mean the renaming to symmetrize, but not the rest.

I agree this takes Mat3 as inputs, but I'm not sure we should aim for a general `symmetrize` function that deduces what it should do by dispatch, since we generally don't use types a lot (eg a density is just an array). And if we keep explicit names there's no reason to type arguments. I do agree about the order though

Reply:
then we should rename to `symmetrize_ρ` or `symmetrize_density`?

Reply:
Yeah. We named it symmetry because rho used to be a special type but now that's gone it should be symmetrize_\rho

Reply:
I agree, stress tensor transforms differently from lattices. I renamed it to `symmetrize_stresses` and changed the argument order.

Reply:
I had another look through the code. I think to rename `symmetrize` -> `symmetrize_real` is more appropriate, since we use it for symmetrising real-space quantities.

Reply:
Hm, real is a bit generic... I think we already use "density" quite generically to denote "density or potential" at some other places in the code. I think it's better to stick with "density" even though it's more general, unless we can come up with another name to mean "real-space quantity"?

Reply:
"densitylike" or "ρlike" ??

Reply:
at that point it's simpler to call it density, no ? :-p

Reply:
ok. `symmetrize_density` it is. @jaemolihm Sorry about that bikesheding. If you want do the renaming, otherwise I'll do it in a followup.

Reply:
I'll leave it as your followup. 😄